libs/wingsdk/src/std/resource.ts

```diff
@@ -176,6 +176,33 @@ export abstract class Resource extends Construct implements IResource {
 
   private readonly onLiftMap: Map<IInflightHost, Set<string>> = new Map();
 
+  constructor(scope: Construct, id: string) {
```

Comment:
Can we implement this without adding a dependency on `std.Resource`? We have some plans to deprecate this class (see https://github.com/winglang/wing/issues/3717)

Reply:
we'll need to attach it to whatever class that replaces that- it has to be a parent class for all of the resources (even if it will be empty besides this constructor) should I add a comment?

Reply:
> we'll need to attach it to whatever class that replaces that

You can safely assume every preflight class extends `constructs.Construct`.

Maybe it's possible to inject this logic in the App's `new` function? see here: https://github.com/winglang/wing/blob/cac0b1d1cf9d9c864da1f8ade48e1eeaf3720814/libs/wingsdk/src/core/app.ts#L203-L218